test/e2e/agentgateway_e2e_test.go

```diff
@@ -124,4 +135,111 @@ var _ = Describe("Agent Gateway", Ordered, func() {
 		Expect(err).NotTo(HaveOccurred(), "Failed to unmarshal agent card response")
 		Expect(responseMap["url"]).To(Equal(gatewayUrl + "/mocked-agent-exposed-1"))
 	})
+
+	It("should reload when agent is deleted", func() {
+		By("verifying agent is accessible via agent card")
+		_, err := utils.GetRequest(gatewayUrl + "/mocked-agent-exposed-1/.well-known/agent-card.json")
+		Expect(err).NotTo(HaveOccurred(), "Agent card should be accessible before deletion")
+
+		By("stopping port-forward before agent deletion")
+		portForwardCancel()
+		time.Sleep(1 * time.Second) // Allow port-forward to clean up
+
+		By("deleting the agent")
+		err = utils.DeleteAgent("mocked-agent-exposed-1", "default")
+		Expect(err).NotTo(HaveOccurred(), "Failed to delete agent")
+
+		By("waiting for gateway deployment to complete rollout")
```

Comment:
We can probably just remove this, as we wait with Eventually for the 404 status anyways.